helpers/blaze/select.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { A, L, O } from 'ts-toolbelt'
+
+export type LPred<I> = (item: I, pos: number) => boolean
+export type OPred<I> = (item: I, key: string) => boolean
+
+function selectList<L extends L.List, I>(object: L & L.List<I>, pred: LPred<I>): L[number][] {
+  const selected = [] as any[]
+
+  for (let i = 0; i < object.length; ++i) {
+    if (pred(object[i], i)) {
+      selected.push(object[i])
+    }
+  }
+
+  return selected
+}
+
+function selectObject<O extends O.Object, I>(object: O, pred: OPred<I>): O.Optional<O> {
+  const selected = {} as any
+
+  const keys = Object.keys(object)
+  for (let i = 0; i < keys.length; ++i) {
+    if (pred(object[i], keys[i])) {
+      selected[keys[i]] = object[keys[i]]
+    }
+  }
+
+  return selected
+}
+
+/**
+ * Select the subset that matches the predicate.
+ *
+ * @param object to be selected
+ * @param pred to select with
+ * @returns
```

Comment:
**(non-blocking, question)** Are these JS-DOC comments correct or helpful, and does Typescript do this for you? Is this just a convention the codebase follows?

Reply:
The TS docs are correct, I'm not sure I get the point about the helpfulness? Maybe you meant that the docs are not clear enough here? In that case we can always elaborate more about what `select` does, yes. No convention, but we aim to have at least minimal docs for all functions (Here it is pretty much WIP so it does not matter so much).

Reply:
Cool no problems here, was just curios! Thank you for providing detail.